hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -506,10 +534,19 @@ namespace hpx { namespace threads { namespace detail
                 // do background work in parcel layer and in agas
                 if ((scheduler.get_scheduler_mode() & policies::do_background_work) &&
                     num_thread < callbacks.max_background_threads_ &&
-                    !callbacks.background_.empty())
+                    background_thread)
                 {
-                    if (callbacks.background_())
-                        idle_loop_count = 0;
+                    thread_result_type background_result = (*background_thread)();
```

Comment:
Would you see a way to still reset `idle_loop_count` if the background thread has performed any ('useful') work?